BattleTag Item 에 대한 Edit 및 Delete 기능 추가.

happyjem opened the pull request:
BattleTag Item 에 대한 Edit 및 Delete 기능 추가.
제가 제대로 구현 했는지 잘 모르겠습니다. 
잘 보시고 문제가 있으면 수정해주세요.

Unit Test project는 다음에 추가할 생각입니다.

happyjem2012 committed:
Unit Test Project 추가

happyjem2012 committed:
Merge remote-tracking branch 'upstream/master'

happyjem2012 committed:
BattleTag 에 대한 Edit, Delete 기능 추가

happyjem2012 committed:
UnitTest는 공부 후에 다시 추가함

kaki104 committed:
Merge branch 'master' into master

kaki104 commented on Diablo3Hub/Diablo3Hub/ViewModels/BattleTagPageViewModel.cs:
```diff
@@ -120,9 +162,39 @@ private void Init()
+            EditBattleTagCommand = new DelegateCommand<object>( async obj => {
+
+                var battelTag = obj as BattleTag;
+                if (battelTag != null)
```
battelTag -> battleTag 이름 변경해야 할 것 같습니다.
if (battelTag == null) return; 으로 변경해주세요

kaki104 commented on Diablo3Hub/Diablo3Hub/ViewModels/BattleTagPageViewModel.cs:
```diff
@@ -120,9 +162,39 @@ private void Init()
+                }
+                else
+                {
+                    await new Windows.UI.Popups.MessageDialog("내부적인 오류로 인해 Tag Item을 업데이트 할 수 없습니다.").ShowAsync();
```
battelTag가 없으면 특별히 메시지를 출력하지 않아도 될 것 같습니다.

kaki104 commented on Diablo3Hub/Diablo3Hub/ViewModels/BattleTagPageViewModel.cs:
```diff
@@ -120,9 +162,39 @@ private void Init()
+                var battelTag = obj as BattleTag;
+                if (battelTag != null)
+                {
+                    var battleTagDialog = new BattleTagManagementDialog();
```
BattleTagManagementDialog(); 수정을 할 때는 사실 
BattleTagManagementDialog(battleTag); 
이런식으로 생성자를 추가로 만들어서 사용하려고 했습니다. 생성자를 추가로 만들어서 처리 하도록 수정 부탁드립니다.

kaki104 commented on Diablo3Hub/Diablo3Hub/ViewModels/BattleTagPageViewModel.cs:
```diff
@@ -120,9 +162,39 @@ private void Init()
+                    battleTagDialog.ViewModel.SetBattleTag(battelTag);
+
+                    var result = await battleTagDialog.ShowAsync();
+                    if (result != ContentDialogResult.Primary) return;
```
결과가 Primary이라면 화면에 배틀테그 목록을 다시 뿌려 줘야 합니다.

kaki104 commented on Diablo3Hub/Diablo3Hub/ViewModels/BattleTagPageViewModel.cs:
```diff
@@ -120,9 +162,39 @@ private void Init()
+                try
+                {
+                    var result = await DBHelper.Instance.DeleteAsync(obj);
+                    Debug.WriteLine("Delete Row Item Index : {0}", result);
```
삭제를 시키기 전에 확인 창을 출력해서 정말 삭제할지를 물어봐야 할 것 같습니다.
그리고 나서, 삭제를 완료한 후에는 작업을 완료했습니다. 라는 팝업이 출력되는 것이 좋을 것 같습니다.

확인 창으로 사용할만한 녀석이 있는지 모르겠네요..찾아보시고 없다면 저에게 작업을 넘겨주시면..작업하도록 하겠습니다.

happyjem replied:
넵 5가지 수정사항 확인했습니다. 빠른 시일내에 수정하도록 하겠습니다.

kaki104 reviewed (changes requested):
5개의 코맨트를 추가했습니다. 확인 부탁드립니다.

happyjem2012 committed:
VIewModel Locator 추가
BattleTagItemDelConfirmDialog 추가
-- 두개 다 필요한 건지는 잘 모르겠음: 필요없거나 쓸모없으면 지우세요 --
Pull Request 수정 사항 반영
-- Delete 기능 보류 -

happyjem2012 committed:
conflict 수정

happyjem2012 committed:
Conflict 다시 수정

happyjem2012 committed:
Upstream 으로 부터 Conflict 수정

happyjem2012 committed:
conflict 로 인한 재수정

happyjem commented:
반려된 Pull request를 수정하고 나서 다시 pull request 해야 될지 잘 모르겠네요. Visual Studio Git Extension은 편하긴 한데 Sync 의미를 정확하게 이해하지 못하겠네요. 지금 제가 commit 한 내용들이 현재 Pull Request 에 자동으로 반영되지 않으면 새로 Pull Request를 하도록 하겠습니다.  remote 변경된 내용을 반영하지 않고 작업을 해서 Conflict 때문에 정신이 없네요. T.T
